src/dock-api.js

```diff
@@ -99,10 +104,16 @@ export default class DockAPI {
       rpc = Object.assign(rpc, PoaRpcDefs);
     }
 
-    const isWebsocket = this.address && this.address.indexOf('http') === -1;
+    const addressStr = addressArray[0];
```

Comment:
The correct way to handle would be to raise error if a mix of URL types is provided or accept a preference of websocket vs http. This requires a non-trivial amount of work so approving it for now. Can you add a note here please?

Reply:
yeah noticed that too, i was thinking of looking into a way to mix websocket/https urls if possible but im not sure polkadotjs allows this. essentially creating a provider per type/url